stdlib/streams/src/test/resources/test-src/streamingv2-custom-extension-test3.bal

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/runtime;
+import throttler;
+
+public type StockRecord record {
+    string symbol;
+    float price;
+    int volume;
+    int expiryTimestamp?;
+};
+
+public type StockSumRecord record {
+    string symbol;
+    float sumPrice;
+    int volume;
+    int expiryTimestamp;
+};
+
+int index = 0;
+stream<StockRecord> inputStream = new;
+stream<StockSumRecord> outputStream = new;
+StockSumRecord[] globalStockSumArray = [];
+
+function startThrottleQuery() returns (StockSumRecord[]) {
+
+    StockRecord s1  = { symbol: "IBM", price: 700.0, volume: 0 };
+    StockRecord s2  = { symbol: "WSO2", price: 60.5, volume: 1 };
+    StockRecord s3  = { symbol: "IBM", price: 700.0, volume: 0 };
+
+    deployThrottleQuery();
+
+    outputStream.subscribe(function(StockSumRecord e) {printStockDetails(e);});
+
+    inputStream.publish(s1);
+    runtime:sleep(500);
+    inputStream.publish(s2);
+    runtime:sleep(260000);
```

Comment:
can we reduce the sleep time (and also window time) for this test case? If not, the test case alone takes a considerable amount of time.

Reply:
fixed